gog/graph.go

```diff
@@ -75,130 +76,109 @@ func (g *Grapher) Graph(fset *token.FileSet, files []*ast.File, typesPkg *types.
 	g.buildScopeInfo(typesInfo)
 	g.assignPathsInPackage(typesPkg)
 
-	// Accumulate the defs, refs and docs from the package being graphed currently.
-	// If the package is graphed successfully, these are added to Output.
-	v := &astVisitor{g: g, typesPkg: typesPkg, typesInfo: typesInfo}
-
-	v.pkgDefs = append(v.pkgDefs, g.NewPackageDef(filepath.Dir(g.fset.Position(files[0].Package).Filename), typesPkg))
+	g.Defs = append(g.Defs, g.NewPackageDef(filepath.Dir(g.fset.Position(files[0].Package).Filename), typesPkg))
 
 	for _, f := range files {
-		ast.Walk(v, f)
+		ast.Walk(g, f)
 	}
 
 	if !g.SkipDocs {
-		v.pkgDocs = g.emitDocs(files, typesPkg, typesInfo)
+		g.Docs = append(g.Docs, g.emitDocs(files, typesPkg, typesInfo)...)
 	}
-
-	// Transfer pkg graph data to output
-	g.Defs = append(g.Defs, v.pkgDefs...)
-	g.Refs = append(g.Refs, v.pkgRefs...)
-	g.Docs = append(g.Docs, v.pkgDocs...)
 }
 
-type astVisitor struct {
-	g         *Grapher
-	typesPkg  *types.Package
-	typesInfo *types.Info
-
-	pkgDefs []*Def
-	pkgRefs []*Ref
-	pkgDocs []*Doc
-
-	structName string
-}
-
-func (v *astVisitor) Visit(node ast.Node) (w ast.Visitor) {
+func (g *Grapher) Visit(node ast.Node) (w ast.Visitor) {
```

Comment:
Not related to this line, just commenting here

`merged astVisitor into Grapher` why is this a good thing?